graph/formats/rdf/graph.go

```diff
@@ -0,0 +1,510 @@
+// Copyright ©2022 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package rdf
+
+import (
+	"fmt"
+
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/iterator"
+	"gonum.org/v1/gonum/graph/multi"
+	"gonum.org/v1/gonum/graph/set/uid"
+)
+
+// Graph implements an RDF graph satisfying the graph.Graph and graph.Multigraph
+// interfaces.
+type Graph struct {
+	nodes map[int64]graph.Node
+	from  map[int64]map[int64]map[int64]graph.Line
+	to    map[int64]map[int64]map[int64]graph.Line
+	pred  map[int64]map[*Statement]bool
+
+	termIDs map[string]int64
+	ids     *uid.Set
+}
+
+// NewGraph returns a new empty Graph.
+func NewGraph() *Graph {
+	return &Graph{
+		nodes: make(map[int64]graph.Node),
+		from:  make(map[int64]map[int64]map[int64]graph.Line),
+		to:    make(map[int64]map[int64]map[int64]graph.Line),
+		pred:  make(map[int64]map[*Statement]bool),
+
+		termIDs: make(map[string]int64),
+		ids:     uid.NewSet(),
+	}
+}
+
+// addNode adds n to the graph. It panics if the added node ID matches an
+// existing node ID.
+func (g *Graph) addNode(n graph.Node) {
+	if _, exists := g.nodes[n.ID()]; exists {
+		panic(fmt.Sprintf("rdf: node ID collision: %d", n.ID()))
+	}
+	g.nodes[n.ID()] = n
+	g.ids.Use(n.ID())
+}
+
+// AddStatement adds s to the graph. It panics Term UIDs in the statement
```

Comment:
Something (an "if"?) is missing after "It panics".

Reply:
Yes. Fixed.